org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semantictokens/PositionReader.java

```diff
@@ -0,0 +1,170 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Avaloq Evolution AG (http://www.avaloq.com).
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ide.server.semantictokens;
+
+import java.io.IOException;
+import java.io.LineNumberReader;
+import java.io.StringReader;
+
+import org.eclipse.lsp4j.Position;
+
+/**
+ * A position reader inspired by {@link LineNumberReader}.
+ * 
+ * @author Rub�n Porras Campo - Initial Contribution and API
+ * 
+ * @since 2.29
+ */
+public class PositionReader extends StringReader {
```

Comment:
This looks almost like a copy of the GPL licensed LineNumberReader. I'm not an expert about licenses but my gut tells me this is not ok.

Reply:
I think you are right. I will redo this class in a different way.